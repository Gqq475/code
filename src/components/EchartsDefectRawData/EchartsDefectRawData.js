import React from 'react'
import classes from './EchartsDefectRawData.scss'
import Chart from 're-echarts'
import EchartsDefectRawDataTable from 'components/AGTable'

type Props = {
  data: Obejct,
  // getHandlingPath: Function,
};

const option = {
  tooltip: {
    trigger: 'axis'
  },
  legend: {
    data: ['Defect Number', 'Glass Number', 'Defect Rate']
  },
  xAxis: [
    {
      type: 'category',
      data: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']
    }
  ],
  yAxis: [
    {
      type: 'value',
      name: 'Defect Number',
      axisLabel: {
        formatter: '{value}'
      }
    },
    {
      type: 'value',
      name: 'Defect Rate',
      axisLabel: {
        formatter: '{value} %'
      }
    }
  ],
  series: [
    {
      name: 'Defect Number',
      type: 'bar',
      data: [1961, 364, 165, 232, 256, 767, 1356, 1622, 326, 200, 64, 533]
    },
    {
      name: 'Glass Number',
      type: 'bar',
      data: [26, 59, 90, 264, 287, 707, 1756, 1822, 487, 188, 3460, 2453]
    },
    {
      name: 'Defect Rate',
      type: 'line',
      yAxisIndex: 1,
      data: [206, 227, 339, 459, 639, 102, 203, 234, 230, 165, 120, 620]
    }
  ]
}

const field = {
  [_.PATH]: 'path',
  [_.PATHTYPE]: 'PathType',
  [_.DEFECTRATE]: 'DefectRate',
  [_.DEFECTNUMBER]: 'DefectNumber',
  [_.GLASSNUMBER]: 'GlassNumber'
}

export class EchartsDefectRawData extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.state = {

    }
    this.setDataSource = this.setDataSource.bind(this)
  }
  componentWillMount () {
    // this.props.getHandlingPath()
  }
  setDataSource () {

  }
  render () {
    if (this.props.data && this.props.data.handlingPath) {
      const PathType = []
      const defectNumber = []
      const defectRate = []
      const glassNumber = []
      this.props.data.handlingPath.result.forEach((item, i) => {
        PathType.push(item.PathType)
        defectNumber.push(item.DefectNumber)
        defectRate.push(parseFloat(item.DefectRate.substr(0, item.DefectRate.length - 1).replace(',', '')))
        glassNumber.push(item.GlassNumber)
      })
      option.series[0].data = defectNumber
      option.xAxis[0].data = PathType
      option.series[2].data = defectRate
      option.series[1].data = glassNumber
    }
    return (
      <div className={classes['EchartsDefectRawData-container']}>
        <Chart option={option} style={{height: '500px'}} />
        {
          <EchartsDefectRawDataTable
            field={field}
            setDataSource={this.setDataSource}
            // loading={this.props.data.reloadAGgrid}
          />
        }
      </div>
    )
  }
}

export default EchartsDefectRawData
