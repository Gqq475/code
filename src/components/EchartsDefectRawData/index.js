import EchartsDefectRawData from './EchartsDefectRawData'
export default EchartsDefectRawData
