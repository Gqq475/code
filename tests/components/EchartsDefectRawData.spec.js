import React from 'react'
import EchartsDefectRawData from 'components/EchartsDefectRawData/EchartsDefectRawData'

describe('(Component) EchartsDefectRawData', () => {
  it('render normally if handlingPath is not exist', () => {
    const data = {
      handlingPath: {
        result: {
          PathType: 'dgds',
          DefectNumber: 12,
          defectRate: 23,
          glassNumber: 2
        }
      }
    }
    const wrapper = shallow(
      <EchartsDefectRawData {...data} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('render normally if handlingPath is exist', () => {
    const data = {}
    const wrapper = shallow(
      <EchartsDefectRawData {...data} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
