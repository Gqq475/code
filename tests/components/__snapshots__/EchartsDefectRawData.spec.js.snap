// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) EchartsDefectRawData render normally if handlingPath is exist 1`] = `
<div
  className="EchartsDefectRawData-container"
>
  <ECharts
    config={Object {}}
    id="chart"
    lazyUpdate={false}
    notMerge={false}
    option={
      Object {
        "legend": Object {
          "data": Array [
            "Defect Number",
            "Glass Number",
            "Defect Rate",
          ],
        },
        "series": Array [
          Object {
            "data": Array [
              1961,
              364,
              165,
              232,
              256,
              767,
              1356,
              1622,
              326,
              200,
              64,
              533,
            ],
            "name": "Defect Number",
            "type": "bar",
          },
          Object {
            "data": Array [
              26,
              59,
              90,
              264,
              287,
              707,
              1756,
              1822,
              487,
              188,
              3460,
              2453,
            ],
            "name": "Glass Number",
            "type": "bar",
          },
          Object {
            "data": Array [
              206,
              227,
              339,
              459,
              639,
              102,
              203,
              234,
              230,
              165,
              120,
              620,
            ],
            "name": "Defect Rate",
            "type": "line",
            "yAxisIndex": 1,
          },
        ],
        "tooltip": Object {
          "trigger": "axis",
        },
        "xAxis": Array [
          Object {
            "data": Array [
              "1",
              "2",
              "3",
              "4",
              "5",
              "6",
              "7",
              "8",
              "9",
              "10",
              "11",
              "12",
            ],
            "type": "category",
          },
        ],
        "yAxis": Array [
          Object {
            "axisLabel": Object {
              "formatter": "{value}",
            },
            "name": "Defect Number",
            "type": "value",
          },
          Object {
            "axisLabel": Object {
              "formatter": "{value} %",
            },
            "name": "Defect Rate",
            "type": "value",
          },
        ],
      }
    }
    style={
      Object {
        "height": "500px",
      }
    }
  />
  <UniverseTable
    field={
      Object {
        "undefined": "GlassNumber",
      }
    }
    setDataSource={[Function]}
  />
</div>
`;

exports[`(Component) EchartsDefectRawData render normally if handlingPath is not exist 1`] = `
<div
  className="EchartsDefectRawData-container"
>
  <ECharts
    config={Object {}}
    id="chart"
    lazyUpdate={false}
    notMerge={false}
    option={
      Object {
        "legend": Object {
          "data": Array [
            "Defect Number",
            "Glass Number",
            "Defect Rate",
          ],
        },
        "series": Array [
          Object {
            "data": Array [
              1961,
              364,
              165,
              232,
              256,
              767,
              1356,
              1622,
              326,
              200,
              64,
              533,
            ],
            "name": "Defect Number",
            "type": "bar",
          },
          Object {
            "data": Array [
              26,
              59,
              90,
              264,
              287,
              707,
              1756,
              1822,
              487,
              188,
              3460,
              2453,
            ],
            "name": "Glass Number",
            "type": "bar",
          },
          Object {
            "data": Array [
              206,
              227,
              339,
              459,
              639,
              102,
              203,
              234,
              230,
              165,
              120,
              620,
            ],
            "name": "Defect Rate",
            "type": "line",
            "yAxisIndex": 1,
          },
        ],
        "tooltip": Object {
          "trigger": "axis",
        },
        "xAxis": Array [
          Object {
            "data": Array [
              "1",
              "2",
              "3",
              "4",
              "5",
              "6",
              "7",
              "8",
              "9",
              "10",
              "11",
              "12",
            ],
            "type": "category",
          },
        ],
        "yAxis": Array [
          Object {
            "axisLabel": Object {
              "formatter": "{value}",
            },
            "name": "Defect Number",
            "type": "value",
          },
          Object {
            "axisLabel": Object {
              "formatter": "{value} %",
            },
            "name": "Defect Rate",
            "type": "value",
          },
        ],
      }
    }
    style={
      Object {
        "height": "500px",
      }
    }
  />
  <UniverseTable
    field={
      Object {
        "undefined": "GlassNumber",
      }
    }
    setDataSource={[Function]}
  />
</div>
`;
